### DISCUSSION TITLE
Occasional disconnections

### DISCUSSION THREAD

**Discussion by marijapili** [foxglove/community#50] [Q&A]
While using the app we noticed some disconnections. Ocassionally, the issue od "failed to communicate with rosapi" will come up, and it is usually fixed by restarting the websocket. Other times, such as when receiving data about lasers or images, there will be a significant message delay. Did you notice any of those issues earlier and do you have some suggestions on how to prevent it from happening? Do you think that frequency at which certain topics are updated can be managed?

Also, do you support or plan on supporting connections via websocket secure?

**Comment by jtbandes** [comment]
Hi – can you give some more details about your setup? Is your network connection between Studio and Rosbridge stable?

If you are running into bandwidth issues with data transfer, the best way to manage the frequency of certain topics would be to simply reduce the publishing frequency (or re-publish at a downsampled rate) or only subscribe to topics that are less frequent. The Rosbridge server does not offer any settings to reduce the message rate. You could also explore using a [custom websocket server](https://foxglove.dev/docs/studio/connection/foxglove-websocket) to suit your needs.

`wss://...` urls should work in the app. Have you run into issues with them?

**Reply by marijapili** [reply]
The connection is stable, it's a rosbridge connection from a desktop to a robot on the same network. Reducing the publishing rate is not an option at this point as the info published is used elsewhere. I will look into the custom websocket.

Haven't tried the wss yet, just wasn't sure if it was supported. Thank you for your answer!

**Reply by jtbandes** [reply]
Another option for reducing the rate is to use something like [`throttle` from topic_tools](http://wiki.ros.org/topic_tools/throttle) to advertise a new topic that can be visualized in Studio with a reduced rate.
